### PR TITLE
Make __call__() work for Activation layers

### PR DESCRIPTION
**Pull request by farizrahman4u**
As `K.softmax()` do not work with 3D data, I used to do `Y = Activation('softmax')(X)`, treating the temporary layer as a function. Recently this has stopped working due to changes made to `__call__()`.

### CODE REVIEW

File: keras/layers/core.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def __call__(self, X, mask=None, train=False):
         if tmp_previous is not None:
             self.set_previous(tmp_previous, False)
         else:
-            self.clear_previous(False)
+            del self.previous
```
Comment (neggert) [inline keras/layers/core.py]: This will break for container layers. You need to use the `clear_previous` method so that containers can override the inherited implementation.

**Comment by farizrahman4u** [comment]
@neggert Just re-implemented `__call__()` in the `Activation` class :) . Don't you think `__call__()` is getting way too complicated ? With the dummy layer and all..?

**Comment by neggert** [comment]
The point of the dummy layer is that it works for both containers and normal layers. It's a little more complicated in Layer, but it ends up being less complicated overall because child classes don't have to override it.

Along those lines, I don't think overriding `__call__` in `Activation` is the right way to go about this. Actually, all you need to do to get your example working is to give the temporary `Activation` layer an input shape (`Y = Activation('softmax', input_shape=(5, 10))(X)`). This fits with @fchollet's desire that a layer should always either have a previous layer or be registered as an input (See discussion in #1721).

**Comment by farizrahman4u** [comment]
I am actually trying to avoid this.:`Y = Activation('softmax', input_shape=(5, 10))(X)`

**Comment by neggert** [comment]
How about `Y = keras.activations.softmax(X)`?

**Comment by farizrahman4u** [comment]
Oh! Why didn't I think of that. Thanks!
